pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.messaging;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+import org.pkl.core.SecurityManager;
+import org.pkl.core.SecurityManagerException;
+import org.pkl.core.messaging.Messages.ListModulesRequest;
+import org.pkl.core.messaging.Messages.ListModulesResponse;
+import org.pkl.core.messaging.Messages.ReadModuleRequest;
+import org.pkl.core.messaging.Messages.ReadModuleResponse;
+import org.pkl.core.module.ExternalModuleResolver;
+import org.pkl.core.module.PathElement;
+
+public class MessageTransportModuleResolver implements ExternalModuleResolver {
```

Comment:
I don't think this needed to be moved? Package `externalreader` was not exporting `MessageTransport`.

Reply:
I think I get it? I'm still tripping over how Java does things.

The idea here is to ensure that nothing in `org.pkl.core.messaging` ends up in the `org.pkl.core.module`/`resource` public API, right? Does that include the constructor for the resolver classes? Or am I missing something here?

Reply:
Yeah; none of signatures of the public methods (including constructors) were using types from `org.pkl.core.messaging `, unless I missed something.

It's totally fine for private methods use to `org.pkl.core.messaging`.

Reply:
The constructors for the resolver classes take a `MessageTransport` so I these do need to move?

Reply:
Ah! My bad!

I just merged this PR. Can you re-submit the other change as a new PR?